pkg/rpmmd/repository.go

```diff
@@ -58,6 +60,12 @@ func (r *RepoConfig) Hash() string {
 	bpts := func(b *bool) string {
 		return fmt.Sprintf("%T", b)
 	}
+	bptsIgnoreNil := func(b *bool) string {
+		if b == nil {
+			return ""
+		}
```

Comment:
AFAIK, the changes hash will impact only the existing repo cache. If the value is `nil`, then the repo configuration should not change and thus it does not make sense to invalidate the existing cache by generating a different repo hash. Given that `bpts()` always returns `*bool` for any pointer value, which would result in the same hash being constructed regardless of the value being `false` or `true` (which is IMO bad), this should be OK.

It may make sense to revisit this whole hash calculation. :thinking: